### PR TITLE
docs: add Text Area Field docs

### PR DESCRIPTION
**Pull request by didoesdigital**
Closes: https://github.com/cultureamp/kaizen-design-system/issues/706

This PR adds a Text Area page

### CODE REVIEW

File: draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import Do from "docs-components/Do"
+import Dont from "docs-components/Dont"
+
+## Visuals
+
+### UI Kit
+
+<iframe style="border: none;" width="744" height="450" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2Fe3TyAOZKfLg9cA01Iy1Mgm%2FKaizen-Site-embed%3Fnode-id%3D50186%253A40775&chrome=DOCUMENTATION" allowfullscreen></iframe>
+
+## Copy guidelines
```
Comment (katsambarcus) [inline draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx:25]: Should we move Copy guidelines to live below the component guidelines?

File: draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+- For example, “Thanks for everything!”
+
+</Dont>
+</DoAndDontContainer>
+
+### Write in sentence case with minimal punctuation
+
+All text area copy should be in sentence case i.e. the first letter of the first word is capitalized and everything else is in lower case unless it’s a proper noun or feature name. Always err on the side of minimal punctuation, so leave out full stops and commas where possible.
+
```
Comment (dlimiter) [inline draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx:106]: Might we have a section here on resizing i.e. fixed containers vs resizable (when and where), do we forbid horizontal resizing etc.

File: draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+<WhenToUseAndWhenNotToUse>
+<WhenToUse>
+
+* Use a Text Area Field to gather longer text input.
+
+</WhenToUse>
+<WhenNotToUse>
+
+* Avoid gathering input from users that isn't absolutely necessary.
+* For shorter text input, use a [Text Field](/components/text-field) instead.
```
Comment (lindsey-collins) [inline draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx:136]: How does this component work with the Rich Text editor? Should we also include guidelines as to when to use this vs. Rich Text editor, or is this actually considered part of the RTE?

File: draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+### Write in sentence case with minimal punctuation
+
+All text area copy should be in sentence case i.e. the first letter of the first word is capitalized and everything else is in lower case unless it’s a proper noun or feature name. Always err on the side of minimal punctuation, so leave out full stops and commas where possible.
+
+## To keep in mind
+
+* **Optional vs required:**
+    * As a rule of thumb, generally use “ (optional)” to indicate optional fields instead of “`*`” to indicate required fields.
+* **Placeholder text** is temporary and vanishes once a user enters text. Avoid placeholder text that must be deleted by the user. Avoid using placeholder text as a label as the user will lose the context of what field they're filling out as soon as it disappears.
+* For **sizing**, consider likely text input and use existing data where possible.
```
Comment (katsambarcus) [inline draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx:112]: More of an engineering consideration, but text areas files should auto-grow in height as the lines of text increase.

File: draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+* For **right-to-left (RTL)** languages, the layout of a text area is swapped. The validation icons are left-aligned.
+
+### Validation
```
Comment (lindsey-collins) [inline draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx:115]: Do we have guidelines for validation that we could link here? It shouldn't live on this page, but could be useful to link to - when to use negative vs. caution, etc.

File: draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx (hunk 4 above)
Comment (dlimiter) [reply draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx:112]: Does this also mean that text areas should auto size to the size of existing  content on page load?

**Comment by ashleecharetteCA** [comment]
Performance current uses the question as a label, and the description sits above the field, which makes sense in the context of an evaluation. Should we have a text field option for this scenario?

**Comment by didoesdigital** [comment]
> Performance current uses the question as a label, and the description sits above the field, which makes sense in the context of an evaluation. Should we have a text field option for this scenario?

Good catch! This will be addressed by "Prominent" variant of text area field in a subsequent PR.

File: draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+## Visuals
+
+### UI Kit
+
+<iframe style="border: none;" width="744" height="450" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2Fe3TyAOZKfLg9cA01Iy1Mgm%2FKaizen-Site-embed%3Fnode-id%3D50186%253A40775&chrome=DOCUMENTATION" allowfullscreen></iframe>
+
+## To keep in mind
+
+* **Optional vs required:**
```
Comment (ActuallyACat) [inline draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx:27]: Is this extendable to other form inputs? Should we instead move it to this page: https://cultureamp.design/guidelines/forms/

Comment (didoesdigital) [reply draft-packages/form/KaizenDraft/Form/TextAreaField/README.mdx:27]: Added a link to the forms page 👍

**Review by mbylstra — APPROVED** [review]

**Pull request merged**